src/objectExplorer/objectExplorerService.ts

```diff
@@ -80,6 +80,14 @@ export class ObjectExplorerService {
                     self._currentNode = TreeNodeInfo.fromNodeInfo(result.rootNode, result.sessionId,
                         undefined, credentials, nodeLabel);
                 }
+                // make a connection if not connected already
+                const nodeUri = ObjectExplorerUtils.getNodeUri(self._currentNode);
+                if (!this._connectionManager.isConnected(nodeUri) &&
```

Comment:
What scenario is this needed for? It seems odd to me that a session that was just created doesn't have a connection associated with it

Reply:
The object explorer session is different from a normal connection. This can happen foe example if you run `New Query` from a disconnected node.